src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/OneDayFileJsonParser.java

```diff
@@ -139,25 +161,59 @@ public MetricsInfoData process(File inputFile) {
                 if (destinationCountryCode >0) {                            // value of -1 or 0 means invalid code
                     kpisInfoData.addDifferentDestinationCodesSet(destinationCountryCode);}
 
-                // TODO (final delete): sleeping block for debugging proposes only
+                // sleeping block for higher visibility of particular processes duration
                 try {
                     Thread.sleep(50);
                 } catch(InterruptedException ex){
-                    // do stuff
+                    logger.error(ex.getMessage(), ex);
                 }
 
                 Instant endingJsonProcess = Instant.now();
                 kpisInfoData.addJsonProcessingDuration(
                         Duration.between(startingJsonProcess, endingJsonProcess).toMillis() );
+
+                rowsCounter++;
+                if(logger.isDebugEnabled()) {
+                    logger.debug("Successfully finished processing of row no: " + rowsCounter);
+                }
             }
             kpisInfoData.incrementProcessedFilesNumber();
 
-            // final mapping
-            metricsInfoData.setKoToOkRatio(koCallsCounter / (float) okCallsCounter);  // the share of KO result to OK result
+            // final mapping into metricsInfoData
+                // the share of KO result to OK result
+                metricsInfoData.setKoToOkRatio(koCallsCounter / (float) okCallsCounter);
 
         } catch (IOException e) {
-            e.printStackTrace();
+            logger.error(e.getMessage(), e);
```

Comment:
Opět - pokud processing souboru skončí s chybou, která se zachytí tady (A přesně to by se stalo, vzhledem k tomu že pokud se URL nevytvoří v MetricsInfoService.processData(...) tak tam v catch bloku jen vypíšeš error na stdout a směle předáš null do této metody),  tak se opět jen zaloguje exception message, tentokrát už na stderr, ale stejně program pokračuje dál...
Je to takhle správně?

Reply:
done